Database curation (#5)
greenelab/knowledge-graph-review: content/03.text-mining.md

### Constructing Databases and Manual Curation

> Constructing a database is an extensive process that can date back to the early 1990s [@pmid:7584356].

Reviewer: Are these the earliest instances? What about Genbank? Does that meet your criteria for a database?

This process involves gathering relevant text such as journal articles, abstracts, or web-based text.
Each collection of text undergoes the process of tagging and normalizing important keywords to represent entities of interest.
This task is commonly known as named entity recognition. 

> Typically, this task uses a dictionary look-up approach [@doi:10.1093/database/bas037] or a machine learning system [@doi:10.1093/bioinformatics/btw343; @doi:10.1142/9789812776136_0062; @doi:10.1186/1471-2105-12-S8-S5] to identify the keywords of interest.

Reviewer: Is this how it's usually done in practice by curators, which is how I read the sentence. I haven't curated, so I'm not sure if it is.

Following named entity recognition, the next step is to detect sentences that assert a relationship of interest (i.e. relationship extraction).
This task can be accomplished via manual curation, automated approaches or a combination of both.

> Succeeding detection, identified sentences are incorporated as evidence for establishing a relationship between mentioned entities of interest. 

Reviewer: Take another look at this sentence. I think there's an issue with it. Are you describing after something happens or after something happens successfully?

Notable databases have been constructed using an extensive amount of manual curation (Table {@tbl:manual-curated-databases}).
For example, COSMIC [@doi:10.1093/nar/gkw1121] was constructed via a group of domain experts scanning the literature for key cancer related genes.
This database has reached close to 45k entries in 2016 and continues to grow throughout the years [@doi:10.1093/nar/gkw1121].

> Studies have shown that these databases contain relatively precise data, but in low quantities [@doi:10.1038/nmeth1209-860; @doi:10.1038/nmeth.1284; @doi:10.1093/database/bau058].

Reviewer: These are 10 years, 10 years, and 5 years old. I suspect things haven't changed too much since then, but is there a more recent look at this?

This happens because the high publication rate is too much for curators to keep up [@doi:10.1007/s11192-010-0202-z].
Plus, this process isn't completely error free as databases were found to be missing annotations and entries [@doi:10.1093/bioinformatics/btm229].

> These combined findings highlight a significant need for faster and more accurate approaches.

Reviewer: What do you mean by accurate? Precision? Recall? A combination?

There have been attempts to fix this bottleneck via semi-automatic ways [@doi:10.1016/j.jbi.2010.11.001; @doi:10.1093/database/bau067; @doi:10.1093/database/bas010; @doi:10.1093/database/baz068].
One approach compared annotating PubMed queries with and without the help of an automated system [@doi:10.1016/j.jbi.2010.11.001].

> It was found that annotating with pre-annotated queries provided a better speed up than annotating from scratch.

Reviewer: I think you're describing a process where a curator is supplied with a computer-annotated sentence (perhaps via some UI?). It took me a while to realize what this was about. It is probably good to assume your reader has less background and give them more context.

Despite the speed up, there was some bias found in the corrected annotations.

> In cases where a query could follow into multiple categories, majority of the annotators defaulted to the same single category.

Reviewer: ```suggestion
In cases where a query could fall into multiple categories, most annotators defaulted to the same single category.
```

> This bias highlights a reoccurring problem of ambiguous sentences.

Reviewer: Has anyone looked into ways to mitigate this via different designs for these systems? If not, maybe good to note as something helpful for someone to follow-up on.

Some sentences can have multiple implications, which makes manual curation a difficult task.

> Manual curation is still need for relationship extraction approaches.

Reviewer: Typo here, I think.

This process can be used to generate gold standard datasets that automated systems use for calibration [@doi:10.1016/j.jbi.2012.04.004; @doi:10.1016/j.artmed.2004.07.016].

> Plus, approaches can use semi-automatic methods to speed up this arduous process.

Reviewer: Would the approaches use this, or would someone?

> Nevertheless, constructing databases using only manual curation is not sufficient.

Reviewer: Why? Do you mean it's not sufficient to produce databases that cover the entire literature? It seems like it could be sufficient for some purposes.

Databases constructed this way will have a significant amount of missing information, since there is too much data to humanly process.

> Future applications should focus on more automated methods to improve the quality of relationship extraction.

Reviewer: Would you support using exclusively computational extraction to build these resources? From my read of this paragraph, it sounds like yes.

> | Database [Reference] | Short Description |

Reviewer: Are these all constructed exclusively via manual curation? How many entities and relationships do they have, and what types of entities and relationships? Would it be good to have columns for manual curation and computational extraction?

| --- | --- |
| Entrez-Gene [@doi:10.1093/nar/gki031] | NCBI's Gene annotation database that contains information pertaining to genes, gene's organism source, phenotypes etc. | 
| Swiss-Prot [@doi:10.1093/nar/gkg095] | A protein protein interaction database that contains proteomic information. |
| PharmGKB [@doi:10.1093/nar/30.1.163] | A database that contains genetic, phenotypic, and clinical information related to pharmacogenomic studies. |
| COSMIC [@doi:10.1093/nar/gkw1121] | A database that contains high resolution human cancer genetic information. | 
| BioGrid [@doi:10.1093/nar/gks1158] | A database for major model organisms. It contains genetic and proteomic information.| 
| Comparative Toxicogenomics Database [@doi:10.1186/1471-2105-10-326] | A database that contains manually curated chemical-gene-disease interactions and  relationships. |
| [@doi:10.1093/nar/gkr1149] | A death domain superfamily database. | 
| Comprehensive Antibiotic Resistance Database [@doi:10.1093/nar/gkw1004] | Manually curated database that contains information about the molecular basis of antimicrobial resistance.|

> Table. A table of reputable databases that were populated through extensive use of manual curation.{#tbl:manual-curated-databases}

Reviewer: What makes them reputable?

### Constructing Databases and Manual Curation

> Database Construction is an extensive process that can date back all the way to 1956 where the first database contained a protein sequence of the insulin molecule [@doi:10.4172/jcsb.1000081].

Reviewer: Don't think construction needs to be caps. Also not sure what "extensive process" refers to here - it has a long history or it requires detailed efforts? It may be best to just remove to avoid confusion.

This process involves gathering relevant text such as journal articles, abstracts, or web-based text.

> At this point curators can directly scan gathered text and detect relationship asserting sentences sentences (i.e. relationship extraction).

Reviewer: is there a difference between `directly scan` and `read`?

Reviewer: also sentences occurs twice.

> An alternative is to take a semi-automatic approach that involves using text mining systems to speed up the curation process.

Reviewer: Would it be better to describe this as an approach that augments curators? Then you could contrast that with fully-automated strategies, which you discuss later.

We discuss the pros and cons of using manual curation to do relationship extraction and mention databases that extensively use this method to populate their fields.

> Notable databases have been constructed using an extensive amount of manual curation (Table {@tbl:manual-curated-databases}).

Reviewer: does "using an extensive amount of" add something to this sentence? If not, I'd suggest removing it.

For example, COSMIC [@doi:10.1093/nar/gkw1121] was constructed via a group of domain experts scanning the literature for key cancer related genes.

> This database has reached close to 45k entries in 2016 and continues to grow throughout the years [@doi:10.1093/nar/gkw1121].

Reviewer: Could you report the number of current annotations? By the time this is published, 2016 will be four years in the past.

Studies have shown that these databases contain relatively precise data, but in low quantities [@doi:10.1038/nmeth1209-860; @doi:10.1038/nmeth.1284; @doi:10.1093/database/bau058; @doi:10.1093/nar/gku1205; @doi:10.1186/s12859-018-2103-8; @doi:10.1093/database/bax043].
This happens because the high publication rate is too much for curators to keep up [@doi:10.1007/s11192-010-0202-z].

> Plus, this process isn't completely error free as databases were found to be missing annotations and entries [@doi:10.1093/bioinformatics/btm229].

Reviewer: Are missing annotations and entries consistent with or in contrast with the assertion above about precise/low-quantity data? It seems consistent to me, so I'm not sure why this gets a separate mention. Is there something subtle about the difference that I'm missing?

> These combined findings highlight a significant need for approaches that  can scale up to an increasing publication rate and improve overall recall of relevant sentences.

Reviewer: Given the way I read the sentence above, I'm a bit confused about `combined` here. Otherwise it seems fine though. Maybe remove `combined` unless there's something I missed above.

> There have been attempts to speed up the manual curation process via semi-automatic ways [@doi:10.1016/j.jbi.2010.11.001; @doi:10.1093/database/bau067; @doi:10.1093/database/bas010; @doi:10.1145/3269206.3269229; @doi:10.1093/database/baz068; @doi:10.1186/s12859-018-2103-8; @doi:10.1186/s12859-018-2021-9].

Reviewer: Is it fair to describe this as augmenting curators with text mining systems? See above.

First step in this context is to use an automatic system to initally extract sentences from text.
This process filters out irrelevant sentences, which means less text for  curators to sift through.
After the pre-filtering step curators can approve or remove the identified sentences.

> This semi-automatic process was found to speed up the curation process compared to manual approach [@doi:10.1016/j.jbi.2010.11.001; @doi:10.1136/amiajnl-2013-001837].

Reviewer: By how much?

> Despite the speed up, this process is prone to erroneous assumptions and produce bias results.

Reviewer: The grammar here needs to be fixed. I'm also not sure what the erroneous assumptions are based on the context around this.

Automated systems identify large quantities of sentences for commonly occurring relationships, but may miss out on less occurring ones [@doi:10.1016/j.jbi.2010.11.001].

> Plus, these systems could identify ambiguous sentences that even expert curators have a hard time descerning [@doi:10.1016/j.jbi.2010.11.001].

Reviewer: Is this a problem with the underlying text itself or do these systems preferentially extract these? I'm having a bit of trouble with interpreting this sentence.

Given these caveats, a future direction would be creating approaches that can mitigate the relationship bias.

> Futhermore, another possible direction is to look into simplifying sentence to resolve ambiguity issues.

Reviewer: If a human can't simplify the sentence, how could a computer? Perhaps there are ways but this needs a reference or a bit fuller treatment.

> Despite the blantant negatives of manual curation, it is still needed in relationship extraction approaches.

Reviewer: Do you want `blatant` here? It may not be quite the right word, unless you want database curators to be up in arms.

This process can be used to generate gold standard datasets that automated systems use for validation [@doi:10.1016/j.jbi.2012.04.004; @doi:10.1016/j.artmed.2004.07.016].
Furthermore, manual curation can be used during the training process of  automated systems (i.e. active learning) [@doi:10.1007/s11390-012-1306-0].
It is important to remember that manual curation alone is precise, but results in low recall rates [@doi:10.1093/bioinformatics/btm229].

> Future databases should consider initially relying on automated methods to obtain sentences at an acceptable recall level, then incorporate manual curation as a medium to fix or remove irrelevant results.

Reviewer: Is this a traditional use of `medium` in the domain of database curation? I usually think of medium as a mode of communication and not a way to do something.
